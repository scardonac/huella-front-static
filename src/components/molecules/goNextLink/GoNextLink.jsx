//Dependencies
import { useContext } from 'react'
//Context
import { NavigateAppContext } from '../../../context/NavigateAppContext'

export const GoNextLink = ({ left = "120px", top = "30px" }) => {
    const { goNext } = useContext(NavigateAppContext)
    return (
        <div className='relative'>
            <a
                className='text-primary-gris3 underline absolute cursor-pointer'
                style={{ left: left, top: top }}
                onClick={goNext}
            >
                {"Siguiente >"}
            </a>
        </div>
    )
}