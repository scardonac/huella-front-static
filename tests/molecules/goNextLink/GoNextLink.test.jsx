//Dependencies
import React from 'react';
//Testing-library
import { fireEvent, render, screen } from '@testing-library/react';
//utils-test-dom
import { act } from 'react-dom/test-utils';
//Component
import { GoNextLink } from '../../../src/components/molecules/goNextLink/GoNextLink';

//mocks context
jest.mock('../../../src/context/NavigateAppContext', () => ({
    NavigateAppContext: {
        goNext: jest.fn(),
    }
}));
//mocks react-router-dom
jest.mock('react-router-dom', () => ({
    Link: 'Link',
}));
//mocks functionNavigate
const functionNavigate = jest.fn();

describe('Pruebas en <GoNextLink />', () => {

    test('Debe renderizarse correctamente', () => {

        // Mock the useContext hook
        const mockContext = {
            goNext: jest.fn()
        };
        jest.spyOn(React, 'useContext').mockReturnValue(mockContext);

        const { baseElement } = render(<GoNextLink functionNavigate={functionNavigate} />);
        expect(baseElement).toMatchSnapshot();
    });

    test('Debe renderizarse correctamente con estilos', () => {
        render(<GoNextLink style="test" />);

        const fnLink = screen.getByTestId('go_next_link');

        act(() => {
            fireEvent.click(fnLink);
        });

    });

});