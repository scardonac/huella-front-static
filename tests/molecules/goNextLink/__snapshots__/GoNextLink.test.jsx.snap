// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GoNextLink /> Debe renderizarse correctamente 1`] = `
<body>
  <div>
    <div
      class="relative"
    >
      <link
        class="text-primary-gris3 underline absolute"
        data-testid="go_next_link"
        id="go_next_link"
        style="left: 120px; top: 30px;"
        to="#"
      >
        Siguiente &gt;
      </link>
    </div>
  </div>
</body>
`;
